### ISSUE TITLE
App's Login Activity not in Open With list

### ISSUE THREAD

**Issue by rowol** [willblaschko/AlexaAndroid#115]
I’m having trouble getting your sample running.    I have signed it with my key, changed the application id, changed the product id, added an api_key.txt generated from LWA, and refactored the package name from willblaschko to mysticind   (Originally I just changed the name in the manifest as mentioned in issue #17, but this does not build)

The app starts fine (but does not seem to try to log in like the Google Play version), I select Send Audio, the click the microphone...   I get an “Open With” box trying for a login that lists Chrome and Alexa Listens (another app I was looking at) but not AlexaVoiceLibrary.   When I install the sample from Google Play, it does work properly, and AlexaVoiceLibrary shows up in the top of the Open With box.  (and the login works, etc)

I did create a Mobile App product on Amazon with a product_id of AlexaVoiceLibrary ... is that correct?  

Any idea what I may have done wrong?
Do you need any more information?

Here’s a copy of the logcat for that process when I run my built version
[logcat.txt](https://github.com/willblaschko/AlexaAndroid/files/1363936/logcat.txt)

**Comment by orlymino** [comment]
Are you using the keystore to build the debug version of the application? I'm guessing that you are starting the application with gradle/Android Studio. When using the "run" command it uses a debug keystore and not the release one.

If you are not, adding the signing information for the debug buildType as well as the release buildType might help. See the build.gradle file of the app module.
Issue #3 show the parameters for the build.gradle file.

**Comment by rowol** [comment]
Thanks for taking the time to respond.   I am new to using Android Studio+Gradle, but I think I am doing to signing correctly and using the keystore to sign the debug build with the release key?  (for the most part I copied what you had...)   You are correct in that I am using both Run and Debug commands from the Android Studio IDE to install the app/APK on a tablet...    

My $/app/build.gradle:

```
if (file('signing.gradle').exists()) {
    apply from: 'signing.gradle'
}else{
    throw new TaskExecutionException(null, new Exception("Unable to find signing.gradle, please replace Android buildTypes > signing configs with your own configuration."));
}
...

android {
   ...
    buildTypes {
        debug{
            versionNameSuffix " Debug"
            minifyEnabled false
            shrinkResources false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
            signingConfig signingConfigs.releaseConfig
        }
        release {
            minifyEnabled false
            shrinkResources true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
            signingConfig signingConfigs.releaseConfig
        }
    }
```

My $/app/signing.gradle (password substituted for real password):

```
android {
    signingConfigs {
        releaseConfig {
            keyAlias 'GenericAndroidKey'
            keyPassword 'password'
            storeFile file('/home/rowol/AndroidKeyStore/android.jks')
            storePassword 'password'
        }
    }
}
```

I got the MD5 and SHA256 values to generate the Amazon api_key.txt from the generated APK that landed in $/app/build/outputs/apk/app-debug.apk.   I am building the app using the either Build/Build APK or Clean/Rebuild Project.     Am I perhaps doing something dumb with the IDE so that I'm not getting the right key in my APK even though I've set it up in the Gradle build?  (i.e. to I have to do something with the Build/Generate Signed APK menu item, or something like that?)  

Thanks!

**Comment by orlymino** [comment]
Oh, I'm sorry, I didn't read fully your error, only had a quick look at the logs and it is not at all related to that your issue.

Just select Chrome, they're might be a difference in the url used to log in and the one declared in the manifest.

**Comment by rowol** [comment]
I tried using Chrome as you suggest... it opened a page at http://na.account.amazon.com/ap/oa  but when I clicked the Continue button, it says 
`"We're sorry!  An error occurred when we tried to process your request.   Rest assured, we're already working on the problem and expect to resolve it shorty"`

I then changed the activity in the manifest and rebuilt.  After that, I was able to successfully LWA from your sample and Alexa worked.

Thank-you!
